test/testsyncvirtualfiles.cpp

```diff
@@ -46,6 +47,25 @@ void triggerDownload(FakeFolder &folder, const QByteArray &path)
     journal.avoidReadFromDbOnNextSync(record._path);
 }
 
+void markForDehydration(FakeFolder &folder, const QByteArray &path)
+{
+    auto &journal = folder.syncJournal();
+    SyncJournalFileRecord record;
+    journal.getFileRecord(path, &record);
+    if (!record.isValid())
+        return;
+    record._type = ItemTypeVirtualFileDehydration;
+    journal.setFileRecord(record);
+    journal.avoidReadFromDbOnNextSync(record._path);
+}
+
+SyncOptions vfsSyncOptions()
+{
+    SyncOptions options;
+    options._vfs = createVfsFromPlugin(Vfs::WithSuffix, nullptr);
```

Comment:
I bet this is leaking.
Maybe a QSharedPointer should actually be appropriate

Reply:
I didn't mind about that since it's in a test. Do you think it matters?

Reply:
I sometimes use valgrind or other leak detector on tests to make sure the code does not leak.

But this can be fixed later.